hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -852,3 +855,92 @@ func (r NodePoolReconciler) validPlatformConfigCondition(ctx context.Context, no
 	SetStatusCondition(&nodePool.Status.Conditions, *condition)
 	return nil, nil
 }
+
+func (r *NodePoolReconciler) supportedVersionSkewCondition(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) (*ctrl.Result, error) {
+	if hcluster.Status.Version == nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolSupportedVersionSkewConditionType,
+			Status:             corev1.ConditionUnknown,
+			Reason:             hyperv1.NodePoolValidationFailedReason,
+			Message:            "HostedCluster version is not available yet, waiting for version information",
+			ObservedGeneration: nodePool.Generation,
+		})
+		return &ctrl.Result{}, fmt.Errorf("HostedCluster version is not available yet")
+	}
+
+	// Get the control plane version string
+	var controlPlaneVersionStr string
+	if len(hcluster.Status.Version.History) == 0 {
+		// If the cluster is in the process of installation, there is no history
+		// Use the desired version as the control plane version
+		controlPlaneVersionStr = hcluster.Status.Version.Desired.Version
+	} else {
+		// If the cluster is installed or upgrading
+		// Start with the most recent version from history as the default
```

Comment:
nit: update the comment if the intent is to start with the oldest version in history